### ISSUE TITLE
pytorch study 1주차

### ISSUE THREAD

**Issue by hyunsu-yang** [hyunsu-yang/PyTorch#7]
## 강의
- https://www.edwith.org/boostcourse-dl-pytorch/joinLectures/22155
## 목차 
Lab-01-1 Tensor Manipulation 1
Lab-01-2 Tensor Manipulation 2
Lab-02 Linear regression
Lab-03 Deeper Look at GD

**Comment by hyunsu-yang** [comment]
## Lab-01 Tensor Manipulation
- pytorch broadcasting 기능을 사용하면서 발생할 수 있는 문제점은? 
  -  자동으로 크기를 맞춰서 연산을 수행
  - 두 메트릭간의 사이즈가 다를 때 자동으로 맞춰주기 때문에
   계산이 잘못되더라도 에러를 보내주지 않음.  즉  에러 디버깅이 어려움.

- pytorch view 기능은 왜 중요할까? 
  - 원소의 수를 유지하면서 텐서의 크기 변경
  - View 함수를 잘 사용하는 것이 딥러닝의 코딩의  매우 중요한 요소

## Lab-02 Linear regression
- MSE 함수는 언제, 무엇을 목적으로 사용되는가? 
  - 평균 제곱 오차(Mean Squared Error, MSE)
  - 비용 함수(cost function) = 손실 함수(loss function) = 오차 함수(error function)  구하는데 사용
  -  평균 제곱 오차의 값을 최소값으로 만드는 Weight와 bias 를 찾아내기 위해

- 강의에 사용된 예제에서 충분히 정확한 weight 값을 얻기 위해서 for문을 얼마나 수행해야 하는가? 
  - ? epoch을 많이 돌리 수록 정확도가 증가하겠지만.. 학습시간과 trade off 일듯..
  - 학습시간에는 learning rate 값도 영향성을 주기 때문에...

## Lab-03 Deeper Look at GD
- gradient descent의 목적은 무엇인가? 
  -  cost가 가장 최소값을 가지게 하는 W(기울기)를 찾는 일
- gradient descent의 단점을 생각해보세요. 
  - 최적값을 찾아 나가기 위해서 한칸 전진할 때마다 모든 데이터 셋을 넣어주어야 한다는 것
  - 그래서 학습이 굉장히 오래 걸리는 문제가 발생하게 되는 것
